File: Week 01/printAllSubstr.cpp
// write a program that will print all non-empty substrings of that given string
// https://www.geeksforgeeks.org/program-print-substrings-given-string/

#include <bits/stdc++.h>

void find_all_substr(std::string str, int n){
	for(int i = 0; i < n; i++){
		for(int j = 1; j <= n-i; j++){
			// str.substr(starting_point, length)
			std::cout << str.substr(i, j) << std::endl;
		}
	}
}

int main(){
	std::string str;
	std::cin >> str;

	int length = str.length();
	find_all_substr(str, length);
	return 0;
}

File: Week 01/printAllSubstr.py
# write a program that will print all non-empty substrings of that given string
# https://www.geeksforgeeks.org/program-print-substrings-given-string/

def find_all_substr(string : str, length : int):
	for i in range(0, length):
		for j in range(i+1, length+1): # length+1 cause for grave last index
			print(string[i:j])

if __name__ == '__main__':
	string = input()
	length = len(string)
	
	find_all_substr(string, length)